### PR TITLE
[PHX-123] Store CustomerVaultId in Omni

### PR DESCRIPTION
**Pull request by tuliot**
## What is this?
We can now get the token for the payment method from NMI, so now we need that stored in Omni so we can use that payment method from other Omni products. This specific feature covers the injection of the token into Omni

## What did I do?
- Modified the `TransactionResult`, `TransactionRequest`, and `Transaction` objects to be aware of `paymentToken`
- Made `ChipDnaDriver` pass the `paymentToken` to the `TransactionRequest`
- Made `TakeMobileReaderPaymentTransaction` use the `/payment-method/token` route, which stores the token along with the payment method

### CODE REVIEW

File: fattmerchant-ios-sdk/Models/PaymentMethod.swift [hunk 1]
```diff
@@ -26,7 +26,25 @@ public class PaymentMethod: Model {
   public var method: PaymentMethodType?
   public var merchantId: String?
   public var customerId: String?
+
+  /// Wether or not Omni should tokenize this PaymentMethod
```
Comment (victor-riv) [inline fattmerchant-ios-sdk/Models/PaymentMethod.swift]: Another castrated ram 😬

File: fattmerchant-ios-sdk/Cardpresent/Data/TransactionRequest.swift [hunk 2]
```diff
@@ -18,10 +37,18 @@ public struct TransactionRequest {
...
+  ///
+  /// - Parameter amount: The  `Amount` to be collected during the transaction
   public init(amount: Amount) {
     self.amount = amount
   }
 
+  /// Initializes a TransactionRequest with the given amount and explicitly sets the tokenize value
+  /// - Parameters:
+  ///   - amount: The `Amount` to be collected during the transaction
+  ///   - tokenize: A value that dictates wether or not the payment method used in the transaction
```
Comment (victor-riv) [inline fattmerchant-ios-sdk/Cardpresent/Data/TransactionRequest.swift]: Another castrated ram 😬

**Review by victor-riv — APPROVED** [review]

File: fattmerchant-ios-sdk/Models/PaymentMethod.swift (hunk 1 above)
Comment (tuliot) [reply fattmerchant-ios-sdk/Models/PaymentMethod.swift]: Addressed in a3678c4

File: fattmerchant-ios-sdk/Cardpresent/Data/TransactionRequest.swift (hunk 2 above)
Comment (tuliot) [reply fattmerchant-ios-sdk/Cardpresent/Data/TransactionRequest.swift]: Addressed in a3678c4

**Pull request merged**